packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -0,0 +1,626 @@
+///    is 100.0px in the main axis, then the diameter is 200.0. And children
+///    will be evenly laid out between that cylinder's -arcsin(0.5) and
+///    arcsin(0.5) angles.
+///
+///    The cylinder's 0 degree side is always centered in the
+///    [RenderListWheelViewport]. The transformation from **untransformed
+///    plane's viewport painting coordinates** is also done such that the child
+///    in the center of that plane will be mostly untransformed with children
+///    above and below it being transformed more as the angle increases.
+class RenderListWheelViewport
+    extends RenderBox
+    with ContainerRenderObjectMixin<RenderBox, ListWheelParentData>,
+         RenderBoxContainerDefaultsMixin<RenderBox, ListWheelParentData>
+    implements RenderAbstractViewport {
+  /// Creates a [RenderListWheelViewport] which renders children on a wheel.
+  ///
+  /// All arguments must not be null. Optional arguments have reasonable defaults.
+  RenderListWheelViewport({
+    @required ViewportOffset offset,
+    double diameterRatio: defaultDiameterRatio,
+    double perspective: defaultPerspective,
+    @required double itemExtent,
+    bool clipToSize: true,
+    bool renderChildrenOutsideViewport: false,
+    List<RenderBox> children,
+  }) : assert(offset != null),
+       assert(diameterRatio != null),
+       assert(diameterRatio > 0, diameterRatioZeroMessage),
+       assert(perspective != null),
+       assert(perspective > 0),
+       assert(perspective <= 0.01, perspectiveTooHighMessage),
+       assert(itemExtent != null),
+       assert(itemExtent > 0),
+       assert(clipToSize != null),
+       assert(renderChildrenOutsideViewport != null),
+       assert(
+         !renderChildrenOutsideViewport || !clipToSize,
+         clipToSizeAndRenderChildrenOutsideViewportConflict,
+       ),
+       _offset = offset,
+       _diameterRatio = diameterRatio,
+       _perspective = perspective,
+       _itemExtent = itemExtent,
+       _clipToSize = clipToSize,
+       _renderChildrenOutsideViewport = renderChildrenOutsideViewport {
+    addAll(children);
+  }
+
+  /// An arbitrary but aesthetically reasonable default value for [diameterRatio].
+  static const double defaultDiameterRatio = 2.0;
+
+  /// Ar arbitrary but aesthetically reasonable default value for [perspective].
+  static const double defaultPerspective = 0.003;
+
+  static const String diameterRatioZeroMessage = "You can't set a diameterRatio "
+      'of 0. It would imply a cylinder of 0 in diameter in which case nothing '
+      'will be drawn.';
+
+  static const String perspectiveTooHighMessage = 'A perspective too high will '
+      'be clipped in the z-axis and therefore not renderable. Value must be '
+      'between 0 and 0.01.';
+
+  static const String clipToSizeAndRenderChildrenOutsideViewportConflict =
+      'Cannot renderChildrenOutsideViewport and clipToSize since children '
+      'rendered outside will be clipped anyway.';
```

Comment:
These strings need documenting

Reply:
(this is why we usually just copy the strings instead of using constants... :-) )

Reply:
Forgot to cherry-pick it back from part 4 :S

Added